src/Microsoft.ML.PipelineInference/AutoMlEngines/UniformRandomEngine.cs

```diff
@@ -30,12 +30,13 @@ public UniformRandomEngine(IHostEnvironment env)
             : base(env, env.Register("UniformRandomEngine(AutoML)"))
         {}
 
-        public override PipelinePattern[] GetNextCandidates(IEnumerable<PipelinePattern> history, int numberOfCandidates)
+        public override PipelinePattern[] GetNextCandidates(IEnumerable<PipelinePattern> history, int numberOfCandidates,
+            Dictionary<string, ColumnPurpose> columnPurpose = null)
```

Comment:
>columnPurpose [](start = 46, length = 13)

any chance you can move them to base class PipelineOptimizerBase so you don't have to pass them around all the time?

Reply:
For the engines, I have created a field in PipelineOptimizerBase. This reduces some clutter.

Recipes code uses a lot of static classes (with static methods).  For those I have added an additional method 

---
In reply to: [199593664](https://github.com/dotnet/machinelearning/pull/461#discussion_r199593664) [](ancestors = 199593664)